File: ios-app/LaVeinteDigital/Theme/LvdColors.swift
import SwiftUI

/// Identidad visual LVD (mirror de `android-app/.../ui/lvd/LvdColors.kt`).
/// Ningún composable usa hex suelto; todo pasa por estos tokens.
enum LvdColors {
    static let navy = Color(hex: 0x161F32)
    static let blue = Color(hex: 0x2462EA)
    static let background = Color(hex: 0xF7F9FA)
    static let surface = Color(hex: 0xFFFFFF)
    static let surfaceSoft = Color(hex: 0xF3F6F9)
    static let textPrimary = Color(hex: 0x161F32)
    static let textSecondary = Color(hex: 0x5E728C)
    static let textMuted = Color(hex: 0x9DA2AA)
    static let border = Color(hex: 0xD9E1EA)
    static let borderStrong = Color(hex: 0xB9CADF)
    static let info = Color(hex: 0x5F92F1)
    static let success = Color(hex: 0x5FCA8A)
    static let warning = Color(hex: 0xF0C65B)
    static let error = Color(hex: 0xEEAFAA)
    static let scrim = Color.black.opacity(0.45)
}

/// Colores de marca heredados (mirror de `ui/theme/Color.kt`).
enum BrandColors {
    static let bg = Color(hex: 0xF8FAFC)
    static let fg = Color(hex: 0x0F172A)
    static let primary = Color(hex: 0x2563EB)
    static let border = Color(hex: 0xE2E8F0)
    static let muted = Color(hex: 0x64748B)
    static let card = Color(hex: 0xFFFFFF)
    static let accent = Color(hex: 0xF1F5F9)
    static let brandNavy = Color(hex: 0x17324D)
    static let brandBlue = Color(hex: 0x2E4F77)
    static let brandCyan = Color(hex: 0x4DA1A8)
}

File: android-app/app/src/main/java/com/laveintedigital/app/ui/lvd/LvdColors.kt
package com.laveintedigital.app.ui.lvd

import androidx.compose.ui.graphics.Color

/**
 * LVD (La Veinte Digital) — tokens de color v1.
 * Referencia visual: la plataforma web actual (dashboard claro + navy + azul).
 *
 * Regla: NINGÚN composable usa colores hex sueltos. Todo pasa por aquí
 * (o por [LvdSemantic]) para poder cambiar la identidad desde un solo archivo.
 */
object LvdColors {

    // ── Brand ──────────────────────────────────────────────────────────────
    val Navy = Color(0xFF161F32)   // cabeceras, top bars, identidad
    val Blue = Color(0xFF2462EA)   // acciones, CTAs, links, selección

    // ── Fondos ─────────────────────────────────────────────────────────────
    val Background = Color(0xFFF7F9FA)
    val Surface = Color(0xFFFFFFFF)
    val SurfaceSoft = Color(0xFFF3F6F9)

    // ── Texto ──────────────────────────────────────────────────────────────
    val TextPrimary = Color(0xFF161F32)
    val TextSecondary = Color(0xFF5E728C)
    val TextMuted = Color(0xFF9DA2AA)

    // ── Bordes ─────────────────────────────────────────────────────────────
    val Border = Color(0xFFD9E1EA)
    val BorderStrong = Color(0xFFB9CADF)

    // ── Semánticos ─────────────────────────────────────────────────────────
    val Info = Color(0xFF5F92F1)
    val Success = Color(0xFF5FCA8A)
    val Warning = Color(0xFFF0C65B)
    val Error = Color(0xFFEEAFAA)

    // Variantes legibles sobre superficies claras (texto de error/success)
    val ErrorStrong = Color(0xFFB3261E)
    val SuccessStrong = Color(0xFF15803D)

    // ── Aliases funcionales ────────────────────────────────────────────────
    val Primary get() = Blue
    val OnPrimary get() = Color.White
    val Scrim get() = Color(0xFF161F32).copy(alpha = 0.40f)
}

/**
 * Alias semánticos por uso. Preferir estos nombres en los composables.
 */
object LvdSemantic {
    val Primary get() = LvdColors.Blue
    val Background get() = LvdColors.Background
    val Surface get() = LvdColors.Surface
    val SurfaceSoft get() = LvdColors.SurfaceSoft
    val TextPrimary get() = LvdColors.TextPrimary
    val TextSecondary get() = LvdColors.TextSecondary
    val TextMuted get() = LvdColors.TextMuted
    val Border get() = LvdColors.Border
    val BorderStrong get() = LvdColors.BorderStrong
    val Success get() = LvdColors.Success
    val Warning get() = LvdColors.Warning
    val Error get() = LvdColors.Error
    val Info get() = LvdColors.Info
}
